### PR TITLE
Dynamic fixed costs by client DDI in scheduled invoices

### PR DESCRIPTION
**Pull request by cruzccl**
#### Type Of Change 
- [ ] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description
Add DDI criteria in dynamic fixed costs added in https://github.com/irontec/ivozprovider/pull/1847

You may choose the country you want and only DDIs of that country will be taken into account.

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php [hunk 1]
```diff
@@ -17,4 +17,24 @@ public function findOneByDdiE164($ddiE164);
      * @return \Ivoz\Provider\Domain\Model\Ddi\DdiInterface | null
      */
     public function findOneByDdiAndCountry(string $ddi, int $countryId);
+
+    /**
+     * @param int $companyId Company id
+     * @return int Number of DDIs of the given Company
+     */
+    public function countByCompany(int $companyId);
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php:25]: These methods require a test

File: web/admin/application/configs/klear/FixedCostsRelInvoiceSchedulersList.yaml [hunk 2]
```diff
@@ -26,7 +26,11 @@ production:
         order:
           fixedCost: true
           type: true
+          quanityGhost: true
```
Comment (mmadariaga) [inline web/admin/application/configs/klear/FixedCostsRelInvoiceSchedulersList.yaml:29]: quantityGhost

File: web/admin/application/languages/en_US/en_US.po [hunk 3]
```diff
@@ -139,6 +139,10 @@ msgstr[1] "Aliases"
 msgid "All"
 msgstr "All"
 
+#, fuzzy
+msgid "All Countries"
+msgstr "GeoIP allowed countries"
```
Comment (mmadariaga) [inline web/admin/application/languages/en_US/en_US.po:144]: Wrong translation

File: web/admin/application/languages/en_US/en_US.po [hunk 4]
```diff
@@ -545,6 +549,18 @@ msgstr[1] "Clients admins"
 msgid "Client filters"
 msgstr "Client filters"
 
+msgid "Client's DDIs NOT from %s (%s)"
+msgstr "Client's DDIs NOT from %s (%s)"
+
+msgid "Client's DDIs from %s (%s)"
+msgstr "Client's DDIs from %s (%s)"
+
+msgid "Client's DDIs from Any Country"
```
Comment (mmadariaga) [inline web/admin/application/languages/en_US/en_US.po:558]: from any Country

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/admin/application/languages/en_US/en_US.po (hunk 3 above)
Comment (Kaian) [reply web/admin/application/languages/en_US/en_US.po]: Fixed thanks!

File: web/admin/application/languages/en_US/en_US.po (hunk 4 above)
Comment (Kaian) [reply web/admin/application/languages/en_US/en_US.po]: _from any country_ in the end

File: web/admin/application/configs/klear/FixedCostsRelInvoiceSchedulersList.yaml (hunk 2 above)
Comment (Kaian) [reply web/admin/application/configs/klear/FixedCostsRelInvoiceSchedulersList.yaml]: Fixed! :blush:

File: library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php [hunk 5]
```diff
@@ -17,4 +17,24 @@ public function findOneByDdiE164($ddiE164);
...
+
+    /**
+     * @param int $companyId Company id
+     * @param int $countryId Country id
+     * @return int Number of DDIs of the given Company and Country as prefix
+     */
+    public function countByCompanyAndCountry(int $companyId, int $countryId): int;
+
+    /**
+     * @param int $companyId Company id
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php]: Done!

File: library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php [hunk 6]
```diff
@@ -17,4 +17,24 @@ public function findOneByDdiE164($ddiE164);
      * @return \Ivoz\Provider\Domain\Model\Ddi\DdiInterface | null
      */
     public function findOneByDdiAndCountry(string $ddi, int $countryId);
+
+    /**
+     * @param int $companyId Company id
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php]: Done!

File: library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php [hunk 7]
```diff
@@ -17,4 +17,24 @@ public function findOneByDdiE164($ddiE164);
...
+     * @return int Number of DDIs of the given Company
+     */
+    public function countByCompany(int $companyId): int;
+
+    /**
+     * @param int $companyId Company id
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/Ddi/DdiRepository.php]: Done!

File: web/admin/application/languages/en_US/en_US.po [hunk 8]
```diff
@@ -139,6 +139,10 @@ msgstr[1] "Aliases"
 msgid "All"
 msgstr "All"
 
+#, fuzzy
```
[the remark this answers is not in the document]
Comment (Kaian) [reply web/admin/application/languages/en_US/en_US.po]: I don't see any fuzzy right now in files, maybe I pushed updated translations while you reviewed (although I don't think so)

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**